manifests/enterprise/dashboard/package.pp

```diff
@@ -0,0 +1,38 @@
+class sensu::enterprise::dashboard::package {
+
+  if $caller_module_name != $module_name {
+    fail("Use of private class ${name} by ${caller_module_name}")
+  }
+
+  if $::sensu::install_repo and $::sensu::enterprise_dashboard {
```

Comment:
What about breaking this up a bit for the scenario where someone wants to install enterprise ($ensure => present), but hosts the RPM on a private yum server (install_repo == false)?  Instead of $ensure = 'present', $ensure => $::sensu::version I think would be better as well (or if the dashboard version doesn't match the sensu version another parameter).